### PR TITLE
Github actions and Fastlane lanes setup to push new build to TestFlight

### PR DESCRIPTION
**Pull request by itsnaidu**
- Fastlane lanes added
  - `release_for_internal` to push a release build to Internal Testers
  - `release_for_beta` to push a release build to Demo Testers
  - `debug_for_internal` to push a debug build to Internal Testers
 
- Github actions added
  - `deploy_For_Internal` workflow will run when a commit is made into the `develop` branch which will call the Fastlane lane named `release_for_internal` and eventually push a build for `Internal Testers`
  - `deploy_For_Beta` workflow will run when a commit is made into the `main` branch which will call the Fastlane lane named `release_for_beta`, and eventually push a build for `Demo Testers`. It also adds `What's new` and submits the app for beta review which is necessary for a beta release.

### CODE REVIEW

**Comment by itsnaidu** [comment]
`Podfile` and `Pods` need to be updated to simulate these actions.
Also, we need to provide Appstore access to push the build. To do this we can use Github secrets to have the Appstore API keys or store and access the Appstore certificates like Provisioning profile and Signing certificate in a private git repo.

**Comment by itsnaidu** [comment]
@RabidFire  Can you please confirm do we need to push a new build when a PR is raised on `develop`?

File: .github/workflows/deploy_For_Internal.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+          MATCH_DEVELOPMENT_PROVISIONING_PROFILE: '${{ secrets.MATCH_DEVELOPMENT_PROVISIONING_PROFILE }}'
+          MATCH_PASSWORD: '${{ secrets.MATCH_PASSWORD }}'
+          APP_STORE_CONNECT_API_KEY_KEY_ID: '${{ secrets.APP_STORE_CONNECT_API_KEY_KEY_ID }}'
+          APP_STORE_CONNECT_API_KEY_ISSUER_ID: '${{ secrets.APP_STORE_CONNECT_API_KEY_ISSUER_ID }}'
+          APP_STORE_CONNECT_API_KEY_CONTENT: '${{ secrets.APP_STORE_CONNECT_API_KEY_CONTENT }}'
+          TEMP_KEYCHAIN_USER: '${{ secrets.TEMP_KEYCHAIN_USER }}'
+          TEMP_KEYCHAIN_PASSWORD: '${{ secrets.TEMP_KEYCHAIN_PASSWORD }}'
+          INTERNAL_TESTERS_GROUP: '${{ secrets.INTERNAL_TESTERS_GROUP }}'
+          DEMO_TESTERS_GROUP: '${{ secrets.DEMO_TESTERS_GROUP }}'
+          CHANGE_LOG: '${{ secreets.CHANGE_LOG }}'
```
Comment (RabidFire) [inline .github/workflows/deploy_For_Internal.yml]: ```CHANGELOG: '${{ secrets.CHANGELOG }}'```

File: Podfile [hunk 2]
```diff
@@ -6,7 +6,8 @@ target 'DashX Demo' do
   use_frameworks!
 
   # Pods for DashX Demo
-
+  pod "DashX", :git => 'https://github.com/dashxhq/dashx-ios.git', :branch => 'develop'
```
Comment (RabidFire) [inline Podfile]: `pod 'DashX'`

File: fastlane/.env [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+# CI only keys for uploading app to store
+# APP_STORE_CONNECT_API_KEY_KEY_ID=
+# APP_STORE_CONNECT_API_KEY_ISSUER_ID=
+# APPSTORE_CONNECT_API_KEY_CONTENT=
+# TEMP_KEYCHAIN_USER=
+# TEMP_KEYCHAIN_PASSWORD=
+
+# INTERNAL_TESTERS_GROUP=
+# DEMO_TESTERS_GROUP=
+# CHANGE_LOG=
```
Comment (RabidFire) [inline fastlane/.env]: _

File: fastlane/Fastfile [hunk 4]
```diff
@@ -15,45 +15,174 @@
...
+      groups: [
+        INTERNAL_TESTERS_GROUP,
+        DEMO_TESTERS_GROUP
+      ],
+      submit_beta_review: true
+    )
+  end
+
+  desc "Upload Release build to TestFlight"
+  lane :release do |options|
```
Comment (RabidFire) [inline fastlane/Fastfile]: private_lane

**Pull request merged**